# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: docshell/shistory/PSHEntry.ipdl
# Commit: 6ebc7618a5fa
# Full Hash: 6ebc7618a5fadb59585a4d8b9f17eb65e5654cc6
# Author: Anny Gakhokidze <agakhokidze@mozilla.com>
# Date: 2020-03-10 21:44:54
# Regressor Bug: 1546759
# File Overlap Count: 2
# Description:
#   Bug 1546759 - Change nsSHistory::WalkHistoryEntries to walk browsing context tree instead of doc shell tree, r=peterv,nika
#   
#   WalkHistoryEntries function gets called by nsSHistory::CloneAndReplaceChild
#   and nsSHistory::SetChildHistoryEntry recursively, so those have to be moved
#   into the parent process. This eliminates many sync IPC calls.
# ==============================================================================

diff -r ed75364b23c3 -r 6ebc7618a5fa docshell/shistory/PSHEntry.ipdl
--- a/docshell/shistory/PSHEntry.ipdl	Tue Mar 10 11:39:06 2020 +0000
+++ b/docshell/shistory/PSHEntry.ipdl	Tue Mar 10 14:28:22 2020 +0000
@@ -108,6 +108,10 @@
   sync ReplaceChild(PSHEntry newChildEntry) returns (nsresult result);
   async ClearEntry(uint64_t aNewSharedID);
   sync CreateLoadInfo() returns (nsDocShellLoadState loadState);
+  sync Clone() returns (CrossProcessSHEntry result);
+  sync SyncTreesForSubframeNavigation(nullable PSHEntry entry, MaybeDiscardedBrowsingContext BC,
+                                     MaybeDiscardedBrowsingContext ignoreBC)
+                        returns (SwapEntriesDocshellData[] entriesToUpdate);
 
   async UpdateLayoutHistoryState(bool scrollPositionOnly,
                                  nsCString[] keys, PresState[] states);